src/eradius_metrics.erl

```diff
@@ -45,54 +45,54 @@ delete_client({{CName, CIP, undefined}, {SName, SIP, SPort}}) ->
-    [update_request(server, ReqType, Address, Ms) || ReqType <- [Type, total]],
-    update_server_time(last_request, Address).
+    [update_request(server, ReqType, Address, Ms) || ReqType <- [Type, "total"]],
+    update_server_time("last_request", Address).
 
 -spec update_server_response(atom(), atom_address()) -> any().
 update_server_response(Type, Address) ->
-    [update_response(server, ReqType, Address) || ReqType <- [Type, total]].
+    [update_response(server, ReqType, Address) || ReqType <- [Type, "total"]].
 
 -spec update_server_time(last_reset | last_config_reset | last_request, atom_address()) -> any().
 update_server_time(Type, Address) ->
     update_time(server, Type, Address).
 
 -spec update_nas_request(atom(), atom_address_pair(), integer()) -> any().
-update_nas_request(pending, {ServerAddress = {SName, SIP, SPort}, {NID, NIP, _NPort}}, Pending) ->
-    update_server_request(pending, ServerAddress, Pending),
-    update_request(nas, pending, {SName, SIP, SPort, NID, NIP}, Pending);
+update_nas_request("pending", {ServerAddress = {SName, SIP, SPort}, {NID, NIP, _NPort}}, Pending) ->
+    update_server_request("pending", ServerAddress, Pending),
+    update_request(nas, "pending", {SName, SIP, SPort, NID, NIP}, Pending);
 update_nas_request(Type, MetricsInfo = {ServerAddress = {SName, SIP, SPort}, {NID, NIP, _NPort}}, Ms) ->
     update_server_request(Type, ServerAddress, Ms),
-    [update_request(nas, ReqType, {SName, SIP, SPort, NID, NIP}, Ms) || ReqType <- [Type, total]],
+    [update_request(nas, ReqType, {SName, SIP, SPort, NID, NIP}, Ms) || ReqType <- [Type, "total"]],
     update_nas_time(last_request, MetricsInfo).
 
 -spec update_nas_response(atom(), atom_address_pair()) -> any().
 update_nas_response(Type, {ServerAddress = {SName, SIP, SPort}, {NID, NIP, _NPort}}) ->
     update_server_response(Type, ServerAddress),
-    [update_response(nas, ReqType, {SName, SIP, SPort, NID, NIP}) || ReqType <- [Type, total]].
+    [update_response(nas, ReqType, {SName, SIP, SPort, NID, NIP}) || ReqType <- [Type, "total"]].
 
 -spec update_nas_time(last_request, atom_address_pair()) -> any().
 update_nas_time(last_request, {{SName, SIP, SPort}, {NID, NIP, _NPort}}) ->
-    update_time(nas, last_request, {SName, SIP, SPort, NID, NIP}).
+    update_time(nas, "last_request", {SName, SIP, SPort, NID, NIP}).
 
 -spec update_client_request(atom(), atom_address_pair(), integer()) -> any().
-update_client_request(pending, {{CName, CIP, undefined}, {SName, SIP, SPort}}, Pending) ->
-    update_request(client, pending, {CName, CIP, SName, SIP, SPort}, Pending);
-update_client_request(retransmission, {{CName, CIP, undefined}, {SName, SIP, SPort}}, Ms) ->
-    update_request(client, retransmission, {CName, CIP, SName, SIP, SPort}, Ms);
+update_client_request("pending", {{CName, CIP, undefined}, {SName, SIP, SPort}}, Pending) ->
+    update_request(client, "pending", {CName, CIP, SName, SIP, SPort}, Pending);
```

Comment:
I don't really like this excessive replacing ot atoms with strings. The conversion can always done later when generating the metrics name.